PerfettoCds/Pipeline/Tables/PerfettoGenericEventTable.cs

```diff
@@ -0,0 +1,218 @@
+            new ColumnMetadata(new Guid("{1aa73a71-1548-44fd-9bcd-854bca78ce2e}"), "Category", "StackID of the event"),
+            new UIHints { Width = 70 });
+
+        private static readonly ColumnConfiguration TypeColumn = new ColumnConfiguration(
+            new ColumnMetadata(new Guid("{01d2b15f-b0fc-4444-a240-0a96f62c2c50}"), "Type", "Type of the event"),
+            new UIHints { Width = 70 });
+
+        public static void BuildTable(ITableBuilder tableBuilder, IDataExtensionRetrieval tableData)
+        {
+            // We dynamically adjust the column headers
+            // This is the max number of fields we can expect for this table
+            int maxFieldCount = Math.Min(AbsoluteMaxFields, tableData.QueryOutput<int>(
+                new DataOutputPath(PerfettoPluginConstants.GenericEventCookerPath, nameof(PerfettoGenericEventCooker.MaximumEventFieldCount))));
+
+            // Get data from the cooker
+            var events = tableData.QueryOutput<ProcessedEventData<PerfettoGenericEvent>>(
+                new DataOutputPath(PerfettoPluginConstants.GenericEventCookerPath, nameof(PerfettoGenericEventCooker.GenericEvents)));
+
+            // Start construction of the column order. Pivot on process and thread
+            List<ColumnConfiguration> allColumns = new List<ColumnConfiguration>() 
+            {
+                ProcessNameColumn,
+                ThreadNameColumn,
+                TableConfiguration.PivotColumn, // Columns before this get pivotted on
+                EventNameColumn,
+                DurationColumn,
+                CategoryColumn,
+                TypeColumn 
+            };
+
+            var tableGenerator = tableBuilder.SetRowCount((int)events.Count);
+            var genericEventProjection = new GenericEventProjection(events);
+
+            // Add all the data projections
+            var processNameColumn = new BaseDataColumn<string>(
+                ProcessNameColumn,
+                genericEventProjection.Compose((genericEvent) => genericEvent.Process));
+            tableGenerator.AddColumn(processNameColumn);
+
+            var threadNameColumn = new BaseDataColumn<string>(
+                ThreadNameColumn,
+                genericEventProjection.Compose((genericEvent) => genericEvent.Thread));
+            tableGenerator.AddColumn(threadNameColumn);
+
+            var eventNameColumn = new BaseDataColumn<string>(
+                EventNameColumn,
+                genericEventProjection.Compose((genericEvent) => genericEvent.EventName));
+            tableGenerator.AddColumn(eventNameColumn);
+
+            var timestampColumn = new BaseDataColumn<Timestamp>(
+                TimestampColumn,
+                genericEventProjection.Compose((genericEvent) => genericEvent.Timestamp));
+            tableGenerator.AddColumn(timestampColumn);
+
+            var durationColumn = new BaseDataColumn<long>(
```

Comment:
Should be TimestampDelta or similar. As long WPA won't be able to use it for duration and won't have automatic support for showing in microseconds, ms, seconds, etc